### PR TITLE
feat: add redstone/garnet chains

### PR DESCRIPTION
**Pull request by holic**
Updated Redstone chain config with proper values and added corresponding Garnet Testnet chain config.

---

## PR-Codex overview
The focus of this PR is to add support for the Garnet chain and update the Redstone chain configurations.

### Detailed summary
- Added `garnet` chain with configurations for Garnet Testnet.
- Updated `redstone` chain with configurations for Redstone.
- Defined source IDs and contract addresses for both chains.
- Updated RPC URLs and block explorer URLs for both chains.

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: src/chains/definitions/garnet.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { chainConfig } from '../../op-stack/chainConfig.js'
+import { defineChain } from '../../utils/chain/defineChain.js'
+
+const sourceId = 17000 // Holesky testnet
+
+export const garnet = defineChain({
+  ...chainConfig,
+  name: 'Garnet Holesky',
```
Comment (holic) [inline src/chains/definitions/garnet.ts:8]:
```suggestion
  name: 'Garnet Testnet',
```

File: src/chains/definitions/garnet.ts
Comment (jxom) [inline src/chains/definitions/garnet.ts:1]: needs export

Comment (holic) [reply src/chains/definitions/garnet.ts:1]: whoops, will add

Comment (holic) [reply src/chains/definitions/garnet.ts:1]: fixed!

**Pull request merged**